### PR TITLE
fix: createSlots に時間帯フィルタを追加し深夜スロット生成を抑制

### PR DESCRIPTION
**Pull request by TITANdayo**
## 問題
scheduleRange（明日〜14日後）を30分刻みで全時間帯のスロットに分割していたため、深夜帯のスロットが大量に生成され、全員が空いている深夜帯がスコア上位を占有し、AIが深夜の候補ばかり提案してしまう不具合があった。
UIの時間帯チェックボックス（朝/昼/夕/夜）はAIプロンプトのテキストにのみ反映され、スロット生成・スコア計算には影響していなかった。

## 変更内容
- **event.ts**: \EVENT_TIME_OF_DAY_CONFIG\ を追加し、時間帯のラベルと時刻範囲(JST)を一元定義。**今回はサーバー側（actions.ts / calculate-free-time-service.ts / schedule-calculator.ts）のみこの定義を参照するよう統一。** UI側（CreateEventForm.tsx / EditEventClient.tsx）は既存のハードコードを維持し、TODOコメントで将来の統一を促す。
- **schedule-calculator.ts**: \createSlots\ に \llowedHourRanges\ パラメータ（配列型）を追加。開始時刻(JST)がいずれかの許可範囲に含まれないスロットは生成段階でスキップする。\calculateTimeRangeScores\ からもパススルーで渡す。
- **calculate-free-time-service.ts**: \calculateFreeTime\ のシグネチャに \llowedHourRanges\ を追加。\slotIntervalMinutes\ は \undefined\ を渡してデフォルト値（30分刻み）を使用。スロット間隔の柔軟化についてはTODOコメントに検討事項を記載。
- **actions.ts**: \EVENT_TIME_OF_DAY_CONFIG\ を参照して \llowedHourRanges\ を構築し \calculateFreeTime\ に渡す。\	imeOfDayCandidate\ はランタイム検証（CONFIG に存在するキーのみ使用）を追加。AIプロンプト用ラベルも同じCONFIGから導出。

### テスト済み（自動テスト）
- \createSlots\ の単体テスト3ケースを追加:
  - 単一時間帯フィルタ（朝のみ）: 許可外スロットが除外されることを確認
  - 複数時間帯フィルタ（朝+夜）: 不連続な範囲が正しく機能することを確認
  - フィルタ未指定: 既存挙動（全スロット生成）が壊れていないことを確認
- 既存テスト含め全7テスト合格

### テスト対象外（手動確認推奨）
- \ctions.ts\ → \calculateFreeTime\ → \calculateTimeRangeScores\ の連結フロー（外部カレンダーAPIのモックが必要なため今回のスコープ外）
- UIチェックボックス → スロット生成への反映（E2Eテスト）

## Summary by CodeRabbit

* **New Features**
  * Schedule suggestions now filter available meeting slots based on selected time-of-day preferences (morning, noon, evening, or night) for more relevant recommendations.

* **Tests**
  * Added comprehensive test coverage for time-of-day filtering functionality in schedule calculations, including multiple time ranges and timezone edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gdsc-osaka/lablink/pull/258?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/app/(auth)/create-event/actions.ts [hunk 1]
```diff
@@ -84,10 +85,24 @@ export async function getScheduleSuggestionsAction(
 
         const scheduleRange = { start: tomorrow, end: rangeEnd };
 
+        // UI の timeOfDayCandidate（morning/noon/evening/night）を
+        // JST の時刻範囲に変換し、スロット生成時のフィルタとして渡す
+        const allowedHourRanges =
+            draft.timeOfDayCandidate.length > 0
+                ? draft.timeOfDayCandidate.map(
+                      (t) => EVENT_TIME_OF_DAY_CONFIG[t].hours,
+                  )
```
Comment (sakura584) [inline src/app/(auth)/create-event/actions.ts]: copilotの指摘の通り検証したほうがいいかもやね
EVENT_TIME_OF_DAY_CONFIG にキーが存在するものだけをフィルタリングするとかでいいと思う！
↑ は一意見なので検証方法はお好きな感じで！

File: src/service/calculate-free-time-service.ts [hunk 2]
```diff
@@ -55,6 +56,15 @@ export const createCalculateFreeTimeService = (
                     eventDurationMinutes,
                     freeSlots,
                     members,
+                    // slotIntervalMinutes: undefined を渡して calculateTimeRangeScores の
+                    // デフォルト値（30分刻み）を使用。
+                    // TODO: 現状は30分固定だが、以下の方法で柔軟化を検討できる:
+                    //   案A: EventDraft に slotIntervalMinutes フィールドを追加し、UIから指定できるようにする
+                    //   案B: イベントの所要時間(eventDurationMinutes)に応じて自動調整する
+                    //        （例: 2時間以上なら60分刻み、30分未満なら15分刻み）
```
Comment (sakura584) [inline src/service/calculate-free-time-service.ts]: 所要時間に合わせて自動調整するのいいね
個人的に案B推しやけど、たいち的にどうやろ？
所要時間短いものに関してスロット刻みすぎるのはあんまいらんかなって思うけど、長時間の予定に合わせてインターバル大きくするのはよさそう

**Review by sakura584 — COMMENTED** [review]
リンタテスト落ちてるから、修正お願いします！
pnpm run fixで自動修正してくれるよ～
CI通ってからマージする感じになります！

File: src/service/calculate-free-time-service.ts (hunk 2 above)
Comment (TITANdayo) [reply src/service/calculate-free-time-service.ts]: なんやかんや希望時間に関わらず30分、もしくは15分刻みくらいで固定で良いような気もしてきました
13:30〜15:30は空いているのに60分刻みだったらスルーされる、みたいな。

Comment (sakura584) [reply src/service/calculate-free-time-service.ts]: たしかにスルーされるのもあるのか
全体ミーティングでも意見聞いてみるか！

**Review by sakura584 — COMMENTED** [review]
LGTMです！
マージお願いします～

**Review by sakura584 — APPROVED** [review]
